### ISSUE TITLE
pollyjs config is undefined when reusing ember build and running ember test

### ISSUE THREAD

**Issue by shankarsridhar** [Netflix/pollyjs#276]
Hello pollyjs contributors and devs,
I am not sure if this is a bug report or a question :)

## TL;DR
pollyjs config from consuming ember app is not available inside the polly js addon methods, when reusing the build, which happens when running the command,
 `ember test --path <build folder name>`

## Description
I am using pollyjs/@ember package for my ember project, and it works just fine, when I run the tests using the command, `ember test`. 

But when I try to reuse an existing build of my ember project(`ember test --path <build folder name. by default, 'dist'>`), I run into this error in command line:

```
Cannot read property 'enabled' of undefined

Stack Trace and Error Report: /var/folders/l3/zfkk52zd2qsdh8lbhdz6gy1r000lf7/T/error.dump.466e5463069e16ef55ccda6970db9a12.log
```

On further looking into the logs and debugging the ember node process, the error was happening in the `testemMiddleware()` method in the [`@pollyjs/ember/index.js`](https://github.com/Netflix/pollyjs/blob/ee25c77f1f469f20754b458b582319947f727c7b/packages/%40pollyjs/ember/index.js#L44) file as show below:

```js
testemMiddleware(app) {
    if (this._config.enabled) { // <== `this._config` is undefined here
      registerExpressAPI(app, this._config.server);
    }
  },
```

And the root cause seems to be that pollyjs `this.app.options.pollyjs` config defined and retrieved from the consuming ember app into the addon, is available only in the `included()` hook; but not available inside the other methods.

### Error Message & Stack Trace

```
stack: TypeError: Cannot read property 'enabled' of undefined
    at Class.testemMiddleware (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/@pollyjs/ember/index.js:45:22)
    at middlewares.forEach.middleware (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/testem/lib/server/index.js:183:9)
    at Array.forEach (<anonymous>)
    at Server.injectMiddleware (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/testem/lib/server/index.js:182:19)
    at Server.createExpress (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/testem/lib/server/index.js:133:10)
    at Server.start (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/testem/lib/server/index.js:41:10)
    at App.startServer (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/testem/lib/app.js:231:24)
    at App.getServer (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/testem/lib/app.js:235:17)
    at Bluebird.using (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/testem/lib/app.js:74:38)
    at tryCatcher (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/bluebird/js/release/util.js:16:23)
    at /Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/bluebird/js/release/using.js:185:26
    at tryCatcher (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/bluebird/js/release/promise.js:517:31)
    at Promise._settlePromise (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/bluebird/js/release/promise.js:574:18)
    at Promise._settlePromise0 (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/bluebird/js/release/promise.js:619:10)
    at Promise._settlePromises (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/bluebird/js/release/promise.js:699:18)
    at Promise._fulfill (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/bluebird/js/release/promise.js:643:18)
    at PromiseArray._resolve (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/bluebird/js/release/promise_array.js:126:19)
    at PromiseArray._promiseFulfilled (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/bluebird/js/release/promise_array.js:144:14)
    at Promise._settlePromise (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/bluebird/js/release/promise.js:579:26)
    at Promise._settlePromise0 (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/bluebird/js/release/promise.js:619:10)
    at Promise._settlePromises (/Users/shsridha/workspace/srd-backstop/gryphon-fe/node_modules/bluebird/js/release/promise.js:699:18)
```

### Config

Defined inside the `ember-cli-build.js`
```js
pollyjs: {
      enabled: EmberApp.env() !== 'production',
      server: {
        expiresIn: null,
        keepUnusedRequests: false
      }
    },
```

## Environment

```
Node.js v10.15.0
darwin 18.7.0
6.4.1
1.15.2
```

**Comment by jasonmit** [comment]
Interesting, I'd expect `included()` to be called.  Unsure if upstream bug or not.

**Comment by shankarsridhar** [comment]
I am not sure if this worked before in my ember project, since I have never tried reusing the ember build in the `dist` folder (by running `ember test --path dist`), until a few weeks back.

My project is using the latest version of @pollyjs/ember.

**Comment by shankarsridhar** [comment]
@jasonmit I've created a repo to demo the problem happening, and how it goes away after removing `@pollyJs/ember`. Hope that helps.

https://github.com/shankarsridhar/reuse-ember-build

^^ ember app uses yarn for dependency management.

**Comment by shankarsridhar** [comment]
On a side note, I ran into a similar issue with one of my addons (that provides custom ember commands), and I moved the config to `environment.js` instead of `ember-cli-build.js` in the consuming ember app. 
After that, I was able to access the config passed on from the consuming ember app.

**Comment by jasonmit** [comment]
@shankarsridhar yeah I see no way around this other than moving the config options to `config/environment.js` or its own config module since the `EmberApp` isn't instantiated in the `ember test --path` flow.

**Comment by shankarsridhar** [comment]
> Interesting, I'd expect `included()` to be called. Unsure if upstream bug or not.

Could be related - https://github.com/ember-cli/ember-cli/issues/8922 ?

**Comment by jasonmit** [comment]
Yup, looks related.  I can work around it but will wait to see how the above progresses first.  Thanks for linking the issues.

**Comment by jasonmit** [comment]
https://github.com/Netflix/pollyjs/pull/277 can you verify this resolves your issue?

**Comment by shankarsridhar** [comment]
> #277 can you verify this resolves your issue?

oh cool. WIll take a shot at it later this weekend.

**Comment by shankarsridhar** [comment]
@jasonmit 
How do I use the fix. Is it part of the latest release version yet ?
Perhaps a certain branch that I should the addon from ?

**Comment by jasonmit** [comment]
We have not yet versioned it but it's on master.

@offirgolan can you release the breaking change?

**Comment by offirgolan** [comment]
@shankarsridhar released under `v3.0.0`
